pkg/api/types.go

```diff
@@ -221,6 +224,25 @@ type GCEPersistentDiskVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// A ISCSI Disk can only be mounted as read/write once.
+type ISCSIDiskVolumeSource struct {
+	// iSCSI target portal
+	Portal string `json:"portal,omitempty" description:"iscsi target portal"`
+	// iSCSI target iqn
+	IQN string `json:"iqn,omitempty" description:"iscsi target iqn"`
+	// iSCSI target lun number
+	Lun int `json:"lun,omitempty" description:"iscsi target lun number"`
+	// Required: Filesystem type to mount.
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs", "ntfs"
+	// TODO: how do we prevent errors in the filesystem from compromising the machine
+	// TODO: why omitempty if required?
```

Comment:
if it is required, don't omitempty.  Is it required or can it be "figured out" ?